packages/adp-tooling/src/types.ts

```diff
@@ -138,3 +142,149 @@ export const enum HttpStatusCodes {
+export type IWriterData<T extends GeneratorType> = IWriter<GeneratorData<T>>;
+export interface IWriter<T extends BaseData> {
+    write(data: T): Promise<void>;
+}
+
+export const enum GeneratorType {
+    ADD_ANNOTATIONS_TO_ODATA = 'Add Annotations to OData',
+    ADD_COMPONENT_USAGES = 'Add Component Usages',
+    ADD_NEW_MODEL = 'Add New Model',
+    CHANGE_DATA_SOURCE = 'Change Data Source',
+    CHANGE_INBOUND = 'Change Inbound'
+}
+
+export const enum GeneratorName {
+    ADD_ANNOTATIONS_TO_ODATA = 'add-annotation-files-generator',
+    ADD_COMPONENT_USAGES = 'add-sapui5-component-usage-generator',
+    ADD_NEW_MODEL = 'add-odata-service-and-model-generator',
+    CHANGE_DATA_SOURCE = 'replace-odata-service-generator',
+    CHANGE_INBOUND = 'change-inbound-generator'
+}
+
+export const enum ChangeTypes {
+    ADD_COMPONENT_USAGES = 'appdescr_ui5_addComponentUsages',
+    ADD_NEW_MODEL = 'appdescr_ui5_addNewModel',
+    ADD_ANOTATIONS_TO_DATA = 'appdescr_app_addAnnotationsToOData',
+    CHANGE_DATA_SOURCE = 'appdescr_app_changeDataSource',
+    ADD_COMPONENT_USAGE_LIBRARY_REFERENCE = 'appdescr_ui5_addLibraries',
+    CHANGE_INBOUND = 'appdescr_app_changeInbound'
+}
+
+export type GeneratorData<T extends GeneratorType> = T extends GeneratorType.ADD_ANNOTATIONS_TO_ODATA
+    ? AnnotationsData
+    : T extends GeneratorType.ADD_COMPONENT_USAGES
+    ? ComponentUsagesData
+    : T extends GeneratorType.ADD_NEW_MODEL
+    ? NewModelData
+    : T extends GeneratorType.CHANGE_DATA_SOURCE
+    ? DataSourceData
+    : T extends GeneratorType.CHANGE_INBOUND
+    ? InboundData
+    : never;
+
+export interface BaseData {
+    projectData: AdpProjectData;
+    timestamp: number;
+}
+
+export interface AnnotationsData extends BaseData {
+    annotationFileName?: string;
+    isInternalUsage: boolean;
+    answers: AnnotationChangeAnswers;
+}
+export interface ComponentUsagesData extends BaseData {
+    answers: ComponentUsagesAnswers;
+}
+export interface NewModelData extends BaseData {
+    answers: NewModelAnswers;
+}
+export interface DataSourceData extends BaseData {
+    answers: DataSourceAnswers;
+    dataSourcesDictionary: { [key: string]: string };
+}
+export interface InboundData extends BaseData {
+    answers: InboundAnswers;
+}
+
+export enum AnnotationFileSelectType {
+    ExistingFile = 0,
+    NewEmptyFile = 1
+}
+
+export interface InboundContent {
+    inboundId: string;
+    entityPropertyChange: { propertyPath: string; operation: string; propertyValue: unknown }[];
+}
+
+export type DataSourceItem = {
+    uri?: string;
+    type: string;
+    settings: {
+        [key: string]: unknown;
+    };
+};
+
+export type PropertyValueType = 'boolean' | 'number' | 'string' | 'binding' | 'object';
+
+export interface AnnotationChangeAnswers {
+    targetODataSource: string;
+    targetAnnotationFileSelectOption: AnnotationFileSelectType;
+    targetAnnotationFilePath: string;
+}
+
+export interface ComponentUsagesAnswers {
```

Comment:
Same as before, something like
```
interface ComponentUsage {
  if: string;
  name: string;
  ...
}
```
is more concise and carries the same informational value

Reply:
Changed in: [3ed676b](https://github.com/SAP/open-ux-tools/pull/1643/commits/3ed676b560a71c69eac49e8258cf84f78c948ade)